lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1250,3 +1256,82 @@ def delete_comment(request, comment_id):
         comment_deleted.send(sender=None, user=request.user, post=cc_comment)
     else:
         raise PermissionDenied
+
+
+@function_trace("_get_course_discussion_user_stats")
+def _get_course_discussion_user_stats(
+    course_key: CourseKey,
+    order_by: UserOrdering = UserOrdering.BY_FLAGS,
+) -> List[Dict[str, Union[str, int]]]:
+    """
+    Get course user discussion stats.
+
+    Args:
+        course_key (CourseKey): course key for course to fetch discussion stats for.
+        order_by (UserOrdering): the ordering to use for sorting data
+
+    Returns:
+        Sorted user stat data for the specified course
+    """
+    course_stats = get_course_user_stats(course_key)
+
+    def key_func(user):
+        key = (user["threads"], user["responses"], user["replies"])
+        if order_by == UserOrdering.BY_FLAGS:
+            key = (user["active_flags"], user["inactive_flags"]) + key
+        return key
+
+    return sorted(course_stats.values(), key=key_func, reverse=True)
+
+
+@function_trace("get_course_discussion_user_stats")
+def get_course_discussion_user_stats(
+    request,
+    course_key_str: str,
+    page: int,
+    page_size: int,
+    order_by: UserOrdering = None,
+) -> Dict:
+    """
+    Get paginated course discussion stats for users in the course.
+
+    Args:
+        request (Request): DRF request
+        course_key_str (str): course key string
+        page (int): Page number to fetch
+        page_size (int): Number of items in each page
+        order_by (UserOrdering): The ordering to use for the user stats
+
+    Returns:
+        Paginated data of a user's discussion stats sorted based on the specified ordering.
+
+    """
+    course_key = CourseKey.from_string(course_key_str)
+    if has_discussion_privileges(user=request.user, course_id=course_key):
+        order_by = order_by or UserOrdering.BY_FLAGS
+    else:
+        order_by = order_by or UserOrdering.BY_ACTIVITY
+        if order_by != UserOrdering.BY_ACTIVITY:
+            raise ValidationError({"order_by": ["Invalid value"]})
+    data = _get_course_discussion_user_stats(course_key, order_by)
+    paginator = Paginator(data, page_size)
+    data = paginator.get_page(page)
+    user_ids = [user["author_id"] for user in data]
+    if not data and page != 1:
+        raise PageNotFoundError("Page not found (No results on this page).")
```

Comment:
nit: This code can be moved right after getting the data from `_get_course_discussion_user_stats`

Reply:
Actually, I think this code is no longer needed. The django paginator will automatically return results from the last page if the page number if too high, and if there are no results it handles that as well.